nucypher/utilities/sandbox/middleware.py

```diff
@@ -58,40 +54,46 @@ def get_certificate(self, host, port, timeout=3, retry_attempts: int = 3, retry_
 
     def consider_arrangement(self, arrangement):
         mock_client = self._get_mock_client_by_ursula(arrangement.ursula)
-        response = mock_client.post("http://localhost/consider_arrangement", bytes(arrangement))
+        response = mock_client.post("http://localhost/consider_arrangement",
+                                    data=bytes(arrangement),
+                                    content_type='application/octet')
         assert response.status_code == 200
         return response
 
     def enact_policy(self, ursula, id, payload):
         mock_client = self._get_mock_client_by_ursula(ursula)
-        response = mock_client.post('http://localhost/kFrag/{}'.format(id.hex()), payload)
+        response = mock_client.post('http://localhost/kFrag/{}'.format(id.hex()), data=payload)
         assert response.status_code == 200
         return True, ursula.stamp.as_umbral_pubkey()
 
     def send_work_order_payload_to_ursula(self, work_order):
         mock_client = self._get_mock_client_by_ursula(work_order.ursula)
         payload = work_order.payload()
         id_as_hex = work_order.arrangement_id.hex()
-        return mock_client.post('http://localhost/kFrag/{}/reencrypt'.format(id_as_hex), payload)
+        return mock_client.post('http://localhost/kFrag/{}/reencrypt'.format(id_as_hex), data=payload)
 
     def get_treasure_map_from_node(self, node, map_id):
         mock_client = self._get_mock_client_by_ursula(node)
-        return mock_client.get("http://localhost/treasure_map/{}".format(map_id))
+        response = mock_client.get("http://localhost/treasure_map/{}".format(map_id))
+        response.content = response.data
+        return response
 
     def node_information(self, host, port, certificate_filepath):
         mock_client = self._get_mock_client_by_port(port)
         response = mock_client.get("http://localhost/public_information")
-        return response
+        if not response.status_code == 200:
+            raise RuntimeError("Or something.")  # TODO: Raise an error here?  Or return False?  Or something?
+        return response.data
 
     def get_nodes_via_rest(self, url, *args, **kwargs):
-        return super().get_nodes_via_rest(url, client=self._get_mock_client_by_url(url), *args, **kwargs)
+        response = super().get_nodes_via_rest(url, client=self._get_mock_client_by_url(url), *args, **kwargs)
+        response.content = response.data  # Little hack for compatibility.
+        return response
 
     def put_treasure_map_on_node(self, node, map_id, map_payload):
         mock_client = self._get_mock_client_by_ursula(node)
-        certificate_filepath = node.certificate_filepath
-
         response = mock_client.post("http://localhost/treasure_map/{}".format(map_id),
-                                    data=map_payload, verify=certificate_filepath)
```

Comment:
Interesting - does the mock Flask client not accept a cert?

Reply:
Err - what is the test client now?

Reply:
Disregard - I found this doc <http://flask.pocoo.org/docs/1.0/testing/#the-testing-skeleton> (RTFD)